src/etc/inc/plugins.inc.d/kea.inc

```diff
@@ -45,6 +45,68 @@ function kea_services()
     return $services;
 }
 
+function kea_run()
+{
+    return [
+        'static_mapping' => 'kea_staticmap',
+    ];
+}
+
+function kea_staticmap($proto = null, $valid_addresses = true, $domain_fallback = 'not.found', $ifconfig_details = null)
+{
+    $staticmap = [];
+
+    if ($proto) {
+        $proto = (int)$proto;
+    }
+
+    if (($proto ?? 4) == 4) {
+        $keav4 = new \OPNsense\Kea\KeaDhcpv4();
+
+        if (!empty((string)$keav4->general->enabled)) {
+            $reservations = $keav4->reservations;
+            $reservation = $reservations->reservation;
+
+            foreach ($reservations->reservation->iterateItems() as $reservation) {
+                $ip_address = (string)$reservation->ip_address;
+                if (empty($ip_address) && $valid_addresses) {
+                    continue;
+                }
+                $hostname = (string)$reservation->hostname;
+                if (empty($hostname)) {
+                    $hostname = null;
+                }
+                $description = (string)$reservation->description;
+                if (empty($description)) {
+                    $description = null;
+                }
+
+                $subnet_node = $keav4->getNodeByReference("subnets.subnet4.{$reservation->subnet}");
+
+                $domain = $domain_fallback;
+                if ($subnet_node) {
+                    if (!empty((string)$subnet_node->option_data->domain_search)) {
+                        $domain = (string)$subnet_node->option_data->domain_search;
```

Comment:
I'm not sure why this is a good idea?

Reply:
I just copied what `dhcpd_staticmap` does, but I guess that only applies to IPv6 it seems.